core/tests/protractor_utils/BlogDashboardPage.js

```diff
@@ -0,0 +1,319 @@
+      currUserProfilePhoto,
+      'Current user profile photo taking too long to display');
+    await waitFor.visibilityOf(
+      currUsername,
+      'Current user name taking too long to display');
+    await action.click('Close Preview', closeBlogCardPreviewButton);
+  };
+
+  this.waitForPublishedBlogPostsToLoad = async function() {
+    await waitFor.visibilityOf(
+      publishedBlogPostsTable, 'Blog posts table taking too long to appear.');
+  };
+
+  this.createNewBlogPost = async function() {
+    await action.click('Create Blog Post button', createBlogPostButton);
+
+    await waitFor.pageToFullyLoad();
+    await waitFor.visibilityOf(
+      BlogPostTitleFieldElement,
+      'Blog Post Editor is taking too long to appear.');
+    return await waitFor.pageToFullyLoad();
+  };
+
+  this.setContent = async function(richTextInstructions) {
+    var schemaBasedEditorTag = element(by.tagName('schema-based-editor'));
+    await waitFor.visibilityOf(
+      schemaBasedEditorTag, 'Schema based editor tag not showing up');
+    var richTextEditor = await forms.RichTextEditor(schemaBasedEditorTag);
+    await richTextEditor.clear();
+    await richTextInstructions(richTextEditor);
+    await action.click('Save Content Button', saveBlogPostContentButton);
+    await waitFor.invisibilityOf(
+      saveBlogPostContentButton,
+      'Content editor takes too long to disappear');
+  };
+
+  this.saveBlogPostAsDraft = async function(
+      blogPostTitle, richTextInstructions) {
+    await action.sendKeys(
+      'New blog post title field', BlogPostTitleFieldElement, blogPostTitle);
+    await this.setContent(richTextInstructions);
+    await waitFor.visibilityOf(
+      blogPostContentDisplay, 'Blog Post content not showing up');
+    await action.click('Save as draft Button', saveBlogPostAsDraftButton);
+    await waitFor.visibilityOfSuccessToast('Blog Post Saved Successfully.');
+  };
+
+  this.publishBlogPost = async function(
+      blogPostTitle, richTextInstructions, tags) {
+    await action.sendKeys(
+      'New blog post title field', BlogPostTitleFieldElement, blogPostTitle);
+    await this.selectTags(tags);
+    await this.setContent(richTextInstructions);
+    await waitFor.visibilityOf(
+      blogPostContentDisplay, 'Blog Post content not showing up');
+    await this.setThumbnailImage();
+    await action.click('Publish Blog Post', publishBlogPostButton);
+    await action.click(
+      'Confirm Publish Blog Post button', confirmButton);
+    await waitFor.visibilityOfSuccessToast(
+      'Blog Post Saved and Published Succesfully.');
+  };
+
+  this.selectTags = async function(tags) {
+    for (let i = 0; i < tags.length; i++) {
+      await action.click(
+        'select blog post tag', blogPostTags.get(tags[i]));
+    }
+  };
+
+  this.navigateToPublishTab = async function() {
+    await waitFor.pageToFullyLoad();
+    await waitFor.visibilityOf(
+      publishTab, 'Unable to find publish tab button');
```

Comment:
This waitFor is redundant with the waitFor in the `action.click` function

Reply:
Removed